doc/specs/#658 - WinGet Download.md

```diff
@@ -0,0 +1,209 @@
+---
+author: Roy MacLachlan RDMaclachlan/roy.maclachlan@microsoft.com
+created on: 2023-02-09
+last updated: 2023-02-09
+issue id: 658
+---
+
+# Spec Title
+
+"For [#658](https://github.com/microsoft/winget-cli/issues/658)"
+
+## Abstract
+
+This spec describes the functionality and high-level implementation design for the downloading of applications using the Windows Package Manager.
```

Comment:
Maybe a few words about "we are targeting downloading 1 installer at a time", downloading multiple may need some more thoughts in future

Reply:
Will add that to the Spec.